<!doctype html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7" lang=""> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8" lang=""> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9" lang=""> <![endif]-->
<!--[if gt IE 8]><!-->
<html class="no-js" lang="en">
<!--<![endif]-->

<head>
	<meta charset="utf-8">
	<title>MERobinson</title>
	<meta name="description" content="">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="icon" type="image/png" href="assets/images/favicon.png">

	<!--Google Font link-->
	<link href="https://fonts.googleapis.com/css?family=Montserrat:400,700" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Raleway:400,600,700" rel="stylesheet">
	<link href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700" rel="stylesheet">

	<link rel="stylesheet" href="assets/css/slick.css">
	<link rel="stylesheet" href="assets/css/slick-theme.css">
	<link rel="stylesheet" href="assets/css/animate.css">
	<link rel="stylesheet" href="assets/css/fonticons.css">
	<link rel="stylesheet" href="assets/css/font-awesome.min.css">
	<link rel="stylesheet" href="assets/css/bootstrap.css">
	<link rel="stylesheet" href="assets/css/magnific-popup.css">
	<link rel="stylesheet" href="assets/css/bootsnav.css">

	<!--For Plugins external css-->
	<!--<link rel="stylesheet" href="assets/css/plugins.css" />-->

	<!--Theme custom css -->
	<link rel="stylesheet" href="assets/css/style.css">
	<!--<link rel="stylesheet" href="assets/css/colors/maron.css">-->

	<!--Theme Responsive css-->
	<link rel="stylesheet" href="assets/css/responsive.css" />

	<script src="assets/js/vendor/modernizr-2.8.3-respond-1.4.2.min.js"></script>
</head>

<body data-spy="scroll" data-target=".navbar-collapse">

	<!-- preloader -->
	<div id="loading">
		<div id="loading-center">
			<div id="loading-center-absolute">
				<div class="object" id="object_one"></div>
				<div class="object" id="object_two"></div>
				<div class="object" id="object_three"></div>
				<div class="object" id="object_four"></div>
			</div>
		</div>
	</div>
	<!--close preloader-->

	<div class="culmn">
		<!--Home page style-->

		<nav class="navbar navbar-light navbar-expand-lg  navbar-fixed white no-background bootsnav">

			<div class="container">

				<!-- Collect the nav links, forms, and other content for toggling -->
				<div class="collapse navbar-collapse" id="navbar-menu">
					<ul class="nav navbar-nav navbar-right" data-in="fadeInDown" data-out="fadeOutUp">
						<li><a href="#hello">Home</a></li>
						<li><a href="#about">About me</a></li>
						<li><a href="#research">Research</a></li>
						<li><a href="#apps">Apps</a></li>
						<li><a href="">Publications</a></li>
						<li><a href="#contact">Contact</a></li>
					</ul>
				</div>
				<!-- /.navbar-collapse -->
			</div>

		</nav>

		<!--open home section-->
		<section id="hello" class="home bg-mega">
			<div class="overlay"></div>
			<div class="container">
				<div class="row">
					<div class="main_home">
						<div class="home_text">
							<h3 class="text-white">Mark E Robinson</h3>
							<h4 class="text-white">Molecular & Computational Biologist</h4>
						</div>
					</div>
				</div>
				<!-- close row-->
			</div>
			<!-- close container -->
		</section>
		<!-- close home section-->
    
    <!--open about section-->
		<section id="about" class="about roomy-50">
			<div class="container">
				<div class="row">
					<div class="main_about">
						<div class="col-md-6">
							<div class="about_content">
								<h2>About me</h2>
								<div class="separator_left"></div>

								<p>As dual wet- and dry-lab scientist during my PhD I established a panel of chromatin-remodeler mutants
								in the developmental model Dictyostelium, performed MNAse-digest sequencing, and developed a novel 
								nucleosome mapping alogithm and visualisation toolset in order to understand the roles of chromatin 
								remodelers in controlling developmental expression programs through nucleosome positioning.</p>
								
								<p>Following my PhD I moved to the lab of Dr. Feldhahn at Imperial College London 
								to gain experience in computational haematology-oncology research. Our work mapping DNA-damage 
								and enhancer reprogramming in transformed B-cells helped to explain why lineage specific markers
								are recurrently mutated in B-cell acute lymphoblastic leukemia (B-ALL). </p>
								
								<p>In 2019 I joined the lab of Dr.Müschen, working together with bench scientists to identify 
								novel therapeutic vulnerabilities of B-cell malignancies. 
								Through integrative analysis of clinical, transcriptional, proteomic and phosphoproteomics data 
								in leukemia & lymphoma I helped uncover an unexpected mechanism of lipid-raft formation
								leading to PI3K amplification loop downstream of the B-cell receptor. 
								Working with Dr. Chan, my identification of mutational segregation between patient cohorts 
								with STAT5- and ERK-pathway driven B-ALL helped lead to the concept of oncogene convergence.
								This work showed that convergence on a single oncogenic driver was essential for development of B-ALL, 
								and that activation of divergent pathways subverts oncogenesis. This convergence theory gives rise 
								to the exciting possibility of synergistically combining divergent pathway activation with principal 
								driver inhibition as a novel therapeutic strategy. By extending this analysis to study all major 
								oncogenic pathways pan-cancer I now hope to identify whether oncogenic convergence is unique to B-ALL, 
								or whether it represents a new hallmark of cancer that can be exploited to design combination therapies.</p>

							</div>
						</div>
						<div class="col-md-6">
							<div class="about_accordion wow fadeIn">
								<div class="content">
								  <h5>Projects</h6>
								</div> 
								<div id="faq_main_content" class="faq_main_content">
									<!-- accordion item-1 -->
									<h6><i class="fa fa-angle-right"></i>Müschen lab - Yale University</h6>
									<div>
										<div class="list-group">
										  <a href="https://www.nature.com/articles/s41586-020-2884-6" class="list-group-item">
										    Ifitm3 as a scaffold for lipid raft formation and PI3K signaling 
										    downstream of the B-cell receptor</a>
										  <a href="https://www.nature.com/articles/s41586-020-2513-4" class="list-group-item">
										    Mutual exclusivity of pathway activating variants in developmental
										    subtypes of B-ALL</a>
								      <a href="https://www.biorxiv.org/content/10.1101/2023.03.13.532152v1" class="list-group-item">
								        Interaction between Ikaros and B-catenin leads to Myc repression
								        and cell death in normal and malignant B-cells</a>
								      <a href="https://www.biorxiv.org/content/10.1101/2023.03.13.532151v1" class="list-group-item">
								        IL2-independent role of CD25 in negative feedback regulation of
								        B-cell receptor signaling</a>
										  <a href="https://medicine.yale.edu/lab/muschen/" class="list-group-item">Lab wesbite</a>
										</div>
									</div>
									<!-- close accordion item-1 -->
                  <!-- accordion item-2 -->
									<h6><i class="fa fa-angle-right"></i> Feldhahn lab - Imperial College London</h6>
									<div>
										<div class="list-group">
										  <a href="https://www.cell.com/cell-reports/fulltext/S2211-1247(17)30114-6" class="list-group-item">
										    Lineage-specific transcriptional programs drive DNA damage and help explain 
										    accumulation of alterations in lineage-related proteins</a>
										  <li class="list-group-item">
										    Chromatin conformation mapping in high risk MECOM-positive AML patients to
										    identify potential crytpic oncogenic enhancers. <i>in preparation</i></li>
										  <a href="https://www.science.org/doi/10.1126/scitranslmed.abf0113" class="list-group-item">
										    Single-cell profiling of human bone marrow progenitors reveals mechanisms of failing erythropoiesis in Diamond-Blackfan anemia</a>
										  <a href="https://www.sciencedirect.com/science/article/pii/S258900422031186X" class="list-group-item">
										    Brd2/4 and Myc regulate alternative cell lineage programmes during early osteoclast differentiation <i>in vitro</i></a>
										  <a href="https://www.imperial.ac.uk/people/n.feldhahn" class="list-group-item">Lab wesbite</a>
										</div>
									</div>
									<!-- close accordion item-2 -->
                  <!-- accordion item-3 -->
									<h6> <i class="fa fa-angle-right"></i> PhD Molecular Biology - Cardiff University </h6>
									<div>
										<div class="list-group">
											<a href="https://orca.cardiff.ac.uk/id/eprint/99875/" class="list-group-item">
											  Exploring the roles of chromatin remodelers in regulating chromatin organisation and transcription in <i>Dictyostelium discoideum</i>
											</a>
										</div>
									</div>
									<!-- close accordion item-3 -->
								</div>
							</div>
						</div>
					</div>
				</div>
				<!-- close row-->
			</div>
			<!-- close container -->
			<br />
		</section>
		<!-- close about section -->
    
		<!--Apps Section-->
		<section id="apps" class="portfolio lightbg">
			<div class="container">
				<div class="row">
					<div class="main_portfolio roomy-30">
						<div class="col-md-8 mr-md-auto ml-md-auto">
							<div class="text-center">
								<h3>APPS</h3>
								<div class="separator_auto"></div>
								<h5>Interacting with, and exploring data is essential to understanding it,
									therefore to communicate data insights to wet-lab scientists I have created a 
									suite of apps to facilitate direct interaction with their results, as well as
									perform simple analyses.
								</a></h5>
							</div>
						</div>
						<div class="row">
							<div class="portfolio_content">
								<div class="row text-center">

									<div class="col-md-3">
										<div class="portfolio_item">
											<a href="http://lymphoblasts.org/functional_studies/" target="_blank">
											<img src="assets/images/hex/functional_studies.png" alt=""/>
											<h6>Vizualize & explore the results of functional studies: RNAseq, proteomics, micro-array</h6>
										</div>
									</div>
									
									<div class="col-md-3">
										<div class="portfolio_item">
											<a href="http://lymphoblasts.org/drug_discovery/" target="_blank">
											<img src="assets/images/hex/drug_discovery.png" alt=""/>
											<h6>Identify cell-type specific therapeutic targets: CRISPR, RNAi & drug screening data</h6>
										</div>
									</div>

									<div class="col-md-3">
										<div class="portfolio_item">
											<a href="http://lymphoblasts.org/cl_phenotypes/" target="_blank">
											<img src="assets/images/hex/cl_phenotypes.png" alt=""/>
											<h6>Molecular phenotypes of cell lines: RNAseq, proteomics & metabolomics data</h6>
										</div>
									</div>

									<div class="col-md-3">
										<div class="portfolio_item">
											<a href="http://lymphoblasts.org/clinical_outcomes/" target="_blank">
											<img src="assets/images/hex/clinical_outcomes.png" alt=""/>
											<h6>Explore the prognostic effects of gene expression: patient micro-array data</h6>
										</div>
									</div>

								</div>
							</div>
						</div>
					</div>
				</div>
				<!-- close row -->
			</div>
			<!-- close container -->
		</section>
		<!-- close section -->

    <!--start research section-->
		<section id="research" class="brand">
			<div class="container">
				<div class="row">
					<div class="main_cbrand roomy-50">
					  <div class="col-md-8 mr-md-auto ml-md-auto">
  					  <div class="text-center">
  								<h3>RESEARCH & SKILLS</h3>
  							  
  							  <div class="separator_auto"></div>
      						<div class="cbrand_item col-md-12">
      						  <h5>Disecting the distinct roles of chromatin remodelers in establishing and maintaining nucleosome positioning and transcriptonal outcomes</h5>
      							<img src="assets/images/fig/nucleosomepos.png" alt="nucleosome summary"></img>
      							<p><b>SUMMARY:</b>   Chromatin remodeling complexes establish nucleosome occupancy, positioning and structure in a
        							dynamic fashion to allow fine-tuning of protein-DNA interactions. The ISWI and CHD families of remodelers possess the ability 
        							to sample DNA linker length between nucleosomes and space nucleosomes evenly in an array. To examine the distinct roles of
        							CHD and ISWI family proteins in the model organism <i>Dictyostelium discoideum</i> I developed a panel of single and double remodeler 
        							knock-out lines and performed MNase-seq, RNA-seq and functional assays. In depth analysis of nucelosome characteristics were 
        							disected by development of a novel bioinformatic algorithm for improved nucleosome positioning quantification and vizualisation.</p>
      						  <p><b>COMPUTATIONAL:</b>   Algorithm development, R package development, peak calling, differential expression, nucleosome positioning, 
      						    NGS raw read processing, R, Python, Perl</p>
      						  <p><b>MOLECULAR:</b>   Cloning, cell culture, RNA-seq, MNAse-seq, Ion-Torrent & Illumina sequencing</p>
      						</div>
      						
      						<div class="separator_auto"></div>
      					  <div class="cbrand_item col-md-12">
      						  <h5>Targeted engagement of β-catenin-Ikaros complexes in refractory B-cell malignancies</h5>
      							<img src="assets/images/fig/bcat.png" alt="bcat ikaros summary"></img>
      							<p><b>SUMMARY:</b>   In most cell types, nuclear β-catenin functions as prominent oncogenic driver and pairs with TCF7-family 
      							factors for transcriptional activation of MYC. Surprisingly, B-lymphoid malignancies not only lacked expression and activating
      							lesions of β-catenin but critically depended on GSK3β for effective β-catenin degradation. Our interactome studies in 
      							B-lymphoid tumors revealed that β-catenin formed repressive complexes with lymphoid-specific Ikaros factors. 
      							Instead of MYC-activation, β-catenin was essential to enable Ikaros-mediated recruitment of nucleosome remodeling and deacetylation (NuRD) 
      							complexes for transcriptional repression of MYC.Preclinical in vivo treatment experiments in patient-derived xenografts validated small 
      							molecule GSK3β-inhibitors for targeted engagement of lymphoid-specific β-catenin-Ikaros complexes as a novel strategy to overcome 
      							conventional mechanisms of drug-resistance in refractory malignancies.</p>
      						  <a href="https://www.biorxiv.org/content/10.1101/2023.03.13.532152v1"><i>Pre-print</i></a>
      						  <p><b>COMPUTATIONAL:</b>   Proteomics protein-interactions, HiChIP chromatin interactions, whole-exome variant calling, 
      						  RNAseq, ChIPseq, evolutionary pressure modeling, dose response</p>
      						</div>
      						
      						<div class="separator_auto"></div>
      						<div class="cbrand_item col-md-12">
      						  <h5>Dynamic phosphatase-recruitment controls B-cell selection and oncogenic signaling</h5>
      							<img src="assets/images/fig/cd25.png" alt="cd25 summary"></img>
      							<p><b>SUMMARY:</b>   Initiation of B-cell receptor (BCR) signaling in normal B-cells, and oncogenic signaling 
      							in B-cell leukemia (B-ALL) and lymphoma both induce high CD25-surface expression. While CD25 is known as an IL2-receptor 
      							chain on T- and NK-cells, the significance of its expression on B-cells was unclear. Our experiments revealed that, 
      							rather than functioning as an IL2-receptor chain, CD25 expressed on B-cells assembled an inhibitory complex including PKCδ 
      							and SHIP1 and SHP1 phosphatases for feedback control of BCR-signaling or its oncogenic mimics. Clinical outcome analysis 
      							demonstrated opposite prognostic effects of CD25 in pre- and post-germinal center malignancies. 
      							Proteomic and phospho-proteomic interactome studies revealed a critical role of CD25 in BCR-feedback regulation:
      							BCR-signaling induced PKCδ-mediated phosphorylation of CD25 on its cytoplasmic tail (S268). 
      							These findings highlight the previously unrecognized role of CD25 in assembling inhibitory phosphatases
      							to control oncogenic signaling in B-cell malignancies and negative selection to prevent autoimmune disease.</p>
      						  <a href="https://www.biorxiv.org/content/10.1101/2023.03.13.532151v1"><i>Pre-print</i></a>
      						  <p><b>COMPUTATIONAL:</b>   Phospho-proteomics, proteomics protein-interactions, RNAseq, ChIPseq</p>
      						  <p><b>MOLECULAR:</b>    Tonsilar B-cell isolation & transduction, DLBCL model development, FACS</p>
      						</div>
      						
      						<div class="separator_auto"></div>
      						<div class="cbrand_item col-md-12">
      						  <h5>Lineage-Specific Genes Are Prominent DNA Damage Hotspots during Leukemic Transformation of B Cell Precursors</h5>
      							<img src="assets/images/fig/lindmg.png" alt="lineage damage summary"></img>
      							<p><b>SUMMARY:</b>   In human leukemia, lineage-specific genes represent predominant targets of deletion, 
      							with lymphoid-specific genes frequently affected in lymphoid leukemia and myeloid-specific genes in myeloid leukemia.
      							To investigate the basis of lineage-specific alterations, we analyzed global DNA damage in primary B cell precursors 
      							expressing leukemia-inducing oncogenes by ChIP-seq. We identified more than 1,000 DNA-damage hotspots, 
      							of which B lineage-specific genes constitute the most prominent targets. We demonstrate that lineage-specific transcription 
      							predisposes lineage-specific genes in transformed B cell precursors to DNA damage, which is likely to promote 
      							the frequent alteration of lineage-specific genes in human leukemia.</p>
      						  <p><b>COMPUTATIONAL:</b>   Differential binding, peak calling, differential expression, genomic sequence analysis, 
      						    feature annotation, feature enrichment, gene set enrichment, R, Python</p>
      						</div>
      						<div class="separator_auto"></div>
      						<div class="cbrand_item col-md-12">
      						  <h5>Signalling input from divergent pathways subverts B cell transformation</h5>
      							<img src="assets/images/fig/staterk.png" alt="stat5-erk summary"></img>
      							<p><b>SUMMARY:</b>   STAT5- and ERK-pathway activating mutations are common in B-cell acute leukemia (B-ALL),
      							and drive proliferation and tumorigenesis. Suprsingly, using mutual exclusivity analysis we found
      							that rather than co-operating to drive leukemogenesis, alterations in STAT5- and ERK-pathways segregated
      							into distinct patient cohorts with significantly less co-occurrence than expected. Experimental follow-up
      						  with single-cell western blot found that in fact, even in the few patient cases exhibiting variant co-occurrence
      						  there was segregation into distinct patient cohorts. Furthermore, forced activation of both pathways simultaneosly 
      						  caused senescence and cell death in vitro, and pharmacological reactivation of the secondary pathway synergized
      						  with primary pathway inhibition in vivo to reduce tumor burden and prolong survival.</p>
      						  <a href="https://www.nature.com/articles/s41586-020-2513-4">Nature 583.7818 (2020): 845-851</a>
      						  <p><b>COMPUTATIONAL:</b>   Patient survival, variant calling & filtering, pathway alteration co-occurrence</p>
      						</div>
      						
      						<div class="separator_auto"></div>
      						<div class="cbrand_item col-md-12">
      						  <h5>Brd2/4 and Myc regulate alternative cell lineage programmes during early osteoclast differentiation <i>in vitro</i></h5>
      							<img src="assets/images/fig/brd.png" alt="brd summary"></img>
      							<p><b>SUMMARY:</b>   Osteoclast (OC) development in response to nuclear factor kappa-Β ligand (RANKL) is critical 
      							for bone homeostasis in health and in disease. Here, we demonstrate that in response to RANKL, 
      							early OC development entails regulation of two alternative cell fate transcriptional programmes, OC vs macrophage. 
      							Both programmes are regulated in a non-redundant manner by increased chromatin binding of Brd2 at 
      							promoters and of Brd4 at enhancers/super-enhancers. Myc, the top RANKL-induced TF, regulates OC development 
      							in co-operation with Brd2/4 and Max and by establishing negative and positive regulatory loops with other lineage-affiliated TFs.
      							These insights into the transcriptional regulation of osteoclastogenesis suggest the clinical potential of selective
      							targeting of Brd2/4 to abrogate pathological OC activation.</p>
      						  <a href="https://www.sciencedirect.com/science/article/pii/S258900422031186X">IScience 24.1 (2021): 101989</a>
      						  <p><b>COMPUTATIONAL:</b>   Differential accessibility, TF-footprinting, gene-regulatory network inference, 
      						  RNA-seq timecourse analysis, super-enhancers</p>
      						</div>
  					  </div>
  					</div>
					</div>
				</div>
				<!--close row -->
			</div>
			<!--close container -->
		</section>
		<!--close research section-->

		<!--start contact section-->
		<section id="contact" class="contact bg-mega fix">
			<div class="container">
				<div class="row">
					<div class="main_contact roomy-100 text-white">
						<div class="col-md-4">
							<div class="rage_widget">
								<div class="widget_head">
									<h3 class="text-white">Connect with me</h3>
									<div class="separator_small"></div>
								</div>
								<p style="color:white;">Connect with me on LinkedIn, GitHub, or contact me at histonemark @ gmail </p>

								<div class="widget_socail m-top-30">
									<ul class="list-inline-item">
										<li><a href="https://www.linkedin.com/in/mark-robinson-68978b113/"><i class="fa fa-linkedin"></i></a></li>
										<li><a href="https://github.com/MERobinson"><i class="fa fa-github"></i></a></li>
									</ul>
								</div>
							</div>
						</div>
					</div>
				</div>
				<!--close row -->
			</div>
			<!--close container -->
		</section>
		<!--close contact section-->

		<!-- scroll up-->
		<div class="scrollup">
			<a href="#"><i class="fa fa-chevron-up"></i></a>
		</div>
		<!-- close scroll up -->

	</div>

	<!-- JS includes -->

	<script src="assets/js/vendor/jquery-1.11.2.min.js"></script>
	<script src="assets/js/vendor/popper.min.js"></script>
	<script src="assets/js/vendor/bootstrap.min.js"></script>
	<script src="assets/js/jquery.magnific-popup.js"></script>
	<script src="assets/js/jquery.easing.1.3.js"></script>
	<script src="assets/js/slick.min.js"></script>
	<script src="assets/js/jquery.collapse.js"></script>
	<script src="assets/js/bootsnav.js"></script>
	<script src="assets/js/plugins.js"></script>
	<script src="assets/js/main.js"></script>

</body>

</html>
